src/fmt.cc

```diff
@@ -0,0 +1,85 @@
+module;
+#ifndef FMT_MODULE_NAME
+#define FMT_MODULE_NAME fmt
+#endif
+// put all implementation-provided headers into the global module fragment
+// to prevent attachment to this module
+#include <cctype>
+#include <cerrno>
+#include <climits>
+#include <cmath>
+#include <cstdarg>
+#include <cstddef>
+#include <cstdint>
+#include <cstdio>
+#include <cstring>
+#include <ctime>
+#include <cwchar>
+
+#include <algorithm>
+#include <chrono>
+#include <exception>
+#include <functional>
+#include <initializer_list>
+#include <iterator>
+#include <limits>
+#include <locale>
+#include <memory>
+#include <ostream>
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <string_view>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+#if _MSC_VER
+#  include <intrin.h>  // _BitScanReverse[64], _BitScanForward[64], _umul128
+#endif
+#ifdef _WIN32
+#  include <io.h>
+#endif
+
+// os.h
+#ifdef FMT_WITH_OS
+  #include <cstdlib>
+  #include <clocale>
+  #if __has_include(<winapifamily.h>)
+    #include <winapifamily.h>
+  #endif
+  #if __has_include(<fcntl.h>)
+    #include <fcntl.h>
+  #endif
+#endif
+
+export module FMT_MODULE_NAME;
+
+#define FMT_MODULE_EXPORT export
+#define FMT_MODULE_EXPORT_BEGIN export {
+#define FMT_MODULE_EXPORT_END }
+
+#define FMT_USE_NONTYPE_TEMPLATE_PARAMETERS 1
```

Comment:
What is this for?

Reply:
C++20 modules unsurprisingly require C++20 which includes NTTPs. `compile.h` enables the use of NTTPs in a non-standard way that excludes compilers like msvc that support this feature. My current code to  support FMT_STRING and FMT_COMPILE relies on this feature. So I enable it by overriding the macro until `compile.h` gets updated to the reality.

Reply:
Looks like we should use this as a fix - https://github.com/foonathan/lexy/issues/15

Reply:
It should be fixed with https://github.com/fmtlib/fmt/pull/2247, but I need help with this PR, some strange MSVC errors...

Reply:
This FMT_STRING thing is bothering me in a different context, too. At present `fmt::to_wstring()` cannot be invoked from client code because the identifier `FMT_STRING` cannot be looked-up from the body of instantiations of `to_wstring`. Neither unqualified lookup nor ADL works 😢. But that should be solved in a different PR.

Reply:
> C++20 modules unsurprisingly require C++20 which includes NTTPs.

While this is true, modules and NTTP are independent features so presence of one doesn't imply presence of the other. I suggest removing this define from the current PR. It can be set when building {fmt}.

Reply:
Ok with me.